packages/components/Badge/src/CounterBadge/CounterBadge.styling.ts

```diff
@@ -44,5 +44,16 @@ export const stylingSettings: UseStylingOptions<CounterBadgeProps, CounterBadgeS
       }),
       ['variant', 'color'],
     ),
+    shadow: buildProps(
+      (tokens: CounterBadgeTokens) => {
+        const { appearance, shadowToken } = tokens;
+        const showShadow = !!shadowToken && (!appearance || appearance === 'filled' || appearance === 'tint');
```

Comment:
Seems like this logic could be in the lookup fucntion?

Reply:
Yeah, thanks for catching that